pkg/operation/cloudbotanist/packetbotanist/machines.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetbotanist
+
+import (
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/operation"
+	"github.com/gardener/gardener/pkg/operation/common"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// GetMachineClassInfo returns the name of the class kind, the plural of it and the name of the Helm chart which
+// contains the machine class template.
+func (b *PacketBotanist) GetMachineClassInfo() (classKind, classPlural, classChartName string) {
+	classKind = "PacketMachineClass"
+	classPlural = "packetmachineclasses"
+	classChartName = "packet-machineclass"
+	return
+}
+
+// GenerateMachineClassSecretData generates the secret data for the machine class secret (except the userData field
+// which is computed elsewhere).
+func (b *PacketBotanist) GenerateMachineClassSecretData() map[string][]byte {
+	return map[string][]byte{
+		machinev1alpha1.PacketAPIKey: b.Shoot.Secret.Data[PacketAPIKey],
+	}
+}
+
+// GenerateMachineConfig generates the configuration values for the cloud-specific machine class Helm chart. It
+// also generates a list of corresponding MachineDeployments. The provided workers will be distributed over
+// the desired facilities. It returns the computed list of MachineClasses and MachineDeployments.
+func (b *PacketBotanist) GenerateMachineConfig() ([]map[string]interface{}, operation.MachineDeployments, error) {
+	var (
+		workers = b.Shoot.Info.Spec.Cloud.Packet.Workers
+		zones   = b.Shoot.Info.Spec.Cloud.Packet.Zones
+
+		machineDeployments = operation.MachineDeployments{}
+		machineClasses     = []map[string]interface{}{}
+	)
+
+	for _, worker := range workers {
+		machineClassSpec := map[string]interface{}{
+			"OS":          b.Shoot.Info.Spec.Cloud.Packet.MachineImage.ID,
+			"projectID":   b.Shoot.Secret.Data[ProjectID],
+			"machineType": worker.MachineType,
+			"facility":    zones,
+			"tags": []string{
+				fmt.Sprintf("kubernetes.io/cluster/%s", b.Shoot.SeedNamespace),
+				"kubernetes.io/role/node",
+			},
+			"secret": map[string]interface{}{
```

Comment:
don't put the secret in the machine class spec - do it like all others: https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/alicloudbotanist/machines.go#L117-L118

Whenever the machine class spec changes a rolling update of the shoot worker nodes is triggered. You don't want to have a rolling update only if the secret changes.

Reply:
OK, got it. Done.